manager/src/main/java/com/redhat/service/smartevents/manager/ProcessorServiceImpl.java

```diff
@@ -156,11 +167,16 @@ public Processor updateProcessor(String bridgeId, String processorId, String cus
         if (!Objects.equals(existingAction, processorRequest.getAction())) {
             throw new BadRequestException("It is not possible to update the Processor's Action.");
         }
+        if (!Objects.equals(existingSource, processorRequest.getSource())) {
+            throw new BadRequestException("It is not possible to update the Processor's Source.");
+        }
 
```

Comment:
@kostola Can you please add a check in here that prevents a user from trying to update the `type` of a `Processor` between `sink` and `source`.

cheers,

Rob

Reply:
Good hint. I introduced an additional check on processor type just to be sure (+ tests).